### PR TITLE
make NEML action block restrictable close #11

### LINKED ISSUE #11: Make NEMLMechanics block restrictable
In some case is necessary to not have NEML working on all block. An example is when lower dimensional elements are present.

### PR DESCRIPTION
**Pull request by arovinelli**
Add the possibility to restrict the action to the specified blocks

### CODE REVIEW

**Comment by reverendbedford** [comment]
Well here's the problem, splitting these up is a nice idea but then you can't rely on new parts elsewhere in the code in each test.  So currently this test fails because I don't yet have the SalehaniIrani3DCTraction object.  Which order am I suppose to run these in?

File: src/actions/NEMLMechanicsAction.C [hunk 1]
```diff
@@ -43,7 +43,9 @@ template <> InputParameters validParams<NEMLMechanicsAction>() {
       "Names of the eigenstrains");
 
   params.addParam<std::vector<SubdomainName>>(
-      "block", "The list of subdomain names where to apply this action");
+      "block", "The list of subdomain names where neml mehcanisc must be used, "
```
Comment (chakra34) [inline src/actions/NEMLMechanicsAction.C]: type -- 'mechanics' and 'physics'

**Comment by chakra34** [comment]
I was wondering whether the above test (example) could be done with NEMLMechanics and some other physics and not the interface (like stress diffusion or something) such that those who do not have the Interface can still use/test this feature. I think if we are doing interface/mechanics coupling, we should first merge/test the interface kernel separately followed by combined and then add this test. I think this is a general feature and could be implemented prior to interface integration.

**Comment by arovinelli** [comment]
@reverendbedford tests should be all independent as I used brand new branches. The `SalehaniIrani3DCTraction` is in moose since christmas, maybe you didn't update recently.
> Well here's the problem, splitting these up is a nice idea but then you can't rely on new parts elsewhere in the code in each test. So currently this test fails because I don't yet have the SalehaniIrani3DCTraction object. Which order am I suppose to merge these in?

**Comment by arovinelli** [comment]
At the present time we are the CZMinterfacekernel that is in MOOSE. 
All tests for it are already in MOOSE.

> I was wondering whether the above test (example) could be done with NEMLMechanics and some other physics and not the interface (like stress diffusion or something) such that those who do not have the Interface can still use/test this feature. I think if we are doing interface/mechanics coupling, we should first merge/test the interface kernel separately followed by combined and then add this test. I think this is a general feature and could be implemented prior to interface integration.

**Comment by arovinelli** [comment]
close because it's stuck. I'll reopen a new one

**Pull request closed without merging**